### PR TITLE
FSF-4515 Planning evaluation implementation

### PR DESCRIPTION
**Pull request by lourenco31**

### CODE REVIEW

File: src/evaluator/evaluator/eufs_adapter.py [hunk 1]
```diff
@@ -78,6 +80,17 @@ def __init__(self, node: rclpy.node.Node):
...
+            [
+                self.node.planning_subscription_,
+                self.node.planning_gt_subscription_,
+            ],
+            10,
+            0.5,
+        )
+
+        self._planning_time_sync_.registerCallback(self.planning_callback)
+
```
Comment (Davide64-dev) [inline src/evaluator/evaluator/eufs_adapter.py:93]: I think this is not necessary as the planning output and the mocker node subscriptions don't need to be in the adapter

File: src/evaluator/evaluator/fsds_adapter.py [hunk 2]
```diff
@@ -63,7 +64,16 @@ def __init__(self, node: rclpy.node.Node):
...
+        self._planning_time_sync_ = message_filters.ApproximateTimeSynchronizer(
+            [
+                self.node.planning_subscription_,
+                self.node.planning_gt_subscription_,
+            ],
+            10,
+            0.5,
+        )
 
+        self._planning_time_sync_.registerCallback(self.planning_callback)
```
Comment (Davide64-dev) [inline src/evaluator/evaluator/fsds_adapter.py:76]: Same thing as EUFS adapter

File: src/evaluator/evaluator/pacsim_adapter.py [hunk 3]
```diff
@@ -61,6 +58,17 @@ def __init__(self, node: rclpy.node.Node):
...
+            [
+                self.node.planning_subscription_,
+                self.node.planning_gt_subscription_,
+            ],
+            10,
+            0.5,
+        )
+
+        self._planning_time_sync_.registerCallback(self.planning_callback)
+
```
Comment (Davide64-dev) [inline src/evaluator/evaluator/pacsim_adapter.py:71]: Same thing

**Review by Davide64-dev — CHANGES_REQUESTED** [review]
The content is great. However, I think the logic of synchronization of the planning and mocker nodes must be on the evaluator class itself, as it does not depend on the logic of the environment it is running on. But what do you think?

**Comment by lourenco31** [comment]
> The content is great. However, I think the logic of synchronization of the planning and mocker nodes must be on the evaluator class itself, as it does not depend on the logic of the environment it is running on. But what do you think?

I did it this way to be coherent with the other nodes and because the callback (self.planning_callback) varies according to the environment, but if there's a solution to the callback problem and you don't mind the different synchronization methods, I'll change it

**Comment by Davide64-dev** [comment]
> > The content is great. However, I think the logic of synchronization of the planning and mocker nodes must be on the evaluator class itself, as it does not depend on the logic of the environment it is running on. But what do you think?
> 
> I did it this way to be coherent with the other nodes and because the callback (self.planning_callback) varies according to the environment, but if there's a solution to the callback problem and you don't mind the different synchronization methods, I'll change it

I think you can change it because the callback logic of the evaluator is already in the evaluator node class. So, I think we might have a repetition.

**Review by Davide64-dev — APPROVED** [review]

**Pull request merged**